src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php

```diff
@@ -128,4 +147,50 @@ private function generateLogoutUrl($key, $referenceType)
 
         return $url;
     }
+
+    /**
+     * Returns the first available logout listener for given context.
+     *
+     * @param string $context The firewall context
+     *
+     * @return array|null The logout listener found or null
+     */
+    private function getListenerForContext($context)
+    {
+        foreach ($this->listeners as $listener) {
+            if (isset($listener[4]) && $context === $listener[4]) {
+                return $listener;
+            }
+        }
+    }
+
+    /**
+     * @param string $key            The firewall key
```

Comment:
I've refactored everything in order to hide the whole listener extraction logic inside `getListener` so this argument isn't necessary anymore.